specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-05-01-privatepreview/Migrations.json

```diff
@@ -0,0 +1,845 @@
+        },
+        "sourceDBServerFullyQualifiedDomainName": {
+          "type": "string"
+        },
+        "targetDBServerFullyQualifiedDomainName": {
+          "type": "string"
+        },
+        "secretParameters": {
+          "$ref": "#/definitions/MigrationSecretParameters"
+        },
+        "dBsToMigrate": {
+          "maxItems": 50,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "migrationResourceGroup": {
+          "$ref": "#/definitions/MigrationResourceGroup"
+        },
+        "setupLogicalReplicationOnSourceDBIfNeeded": {
+          "type": "boolean"
+        },
+        "overwriteDBsInTarget": {
+          "type": "boolean"
+        },
+        "migrationWindowStartTimeInUtc": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "startDataMigration": {
+          "type": "boolean"
+        },
+        "triggerCutover": {
+          "type": "boolean",
+          "description": "To trigger cutover for entire migration we need to send this flag as true"
+        },
+        "DBsToTiggerCutoverOn": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "When you want to trigger cutover for specific databases send triggerCutover flag as true and database names in this array"
+        },
+        "cancel": {
+          "type": "boolean",
+          "description": "To trigger cancel for entire migration we need to send this flag as true"
+        },
+        "DBsToCancelMigrationOn": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "When you want to trigger cancel for specific databases send triggerCutover flag as true and database names in this array"
+        },
+        "migrationMode": {
+          "$ref": "#/definitions/MigrationMode",
+          "description": "There are two types of migration modes Online and Offline"
+        }
+      },
+      "description": "Migration resource properties for patch."
+    },
+    "MigrationResourceGroup": {
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "type": "string"
+        },
+        "subnetResourceId": {
+          "type": "string"
+        }
+      },
+      "description": "Migration resource group."
+    },
+    "MigrationSecretParameters": {
+      "required": [
+        "adminCredentials"
+      ],
+      "type": "object",
+      "properties": {
+        "adminCredentials": {
+          "x-ms-secret": true,
+          "x-ms-external": true,
+          "$ref": "#/definitions/AdminCredentials"
+        }
+      },
+      "description": "Migration secret parameters."
+    },
+    "AdminCredentials": {
+      "type": "object",
+      "required": [
+        "sourceServerPassword",
+        "targetServerPassword"
+      ],
+      "properties": {
+        "sourceServerPassword": {
```

Comment:
we can have a common type for password and use it for both sourceServerPassword and targetServerPassword